yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -66,6 +66,8 @@ private[spark] class Client(
   private val executorMemoryOverhead = args.executorMemoryOverhead // MB
   private val distCacheMgr = new ClientDistributedCacheManager()
   private val isClusterMode = args.isClusterMode
+  private val fireAndForget = isClusterMode &&
+    sparkConf.getBoolean("spark.yarn.cluster.quiet", false)
```

Comment:
What do you think of spark.yarn.client.waitForCompletion?  Not a strong preference, but spark.yarn.waitForCompletion is a little ambiguous on who is waiting for who.